### PR TITLE
Moving initActionConfig to pkg

### LINKED ISSUE #6333: Add function to create configuration for actions in pkg
I'm playing around with the v3 beta, and trying to refactor the terraform provider. There doesn't seem to be an easy way to create a configuration, and I feel like I am duplicating a bunch of logic from initActionConfig in cmd/helm. It would be super helpful if there some methods added to build the configuration object in pkg

### PR DESCRIPTION
**Pull request by aaronmell**
Moved initActionConfig to pkg to make it easier to instantiate the configuration for projects consuming helm

closes #6333 

Signed-off-by: Aaron Mell <amell@lumindigital.com>

**What this PR does / why we need it**:
I was trying migrate the terraform helm provider to v3 beta. Realized that I was basically copying most of the logic in cmd/helm/helm.go to instantiate the configuration. Figured other people would have to do the same, and it should be easier to create the config without knowing about the internals of helm. 

**Special notes for your reviewer**:
I'm pretty new to go, so I'd appreciate candid feedback.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by aaronmell** [comment]
Hmm, I get a different result when I run the tests locally.

**Comment by aaronmell** [comment]
My second commit seems a bit much. I'm not sure if there was a better way to handle the change that I made.

**Comment by thomastaylor312** [comment]
Hey @aaronmell thanks for the PR! So I haven't done a full review yet, but the refactor generally looks good and I'm super happy to see some feedback around using Helm as a library. My concern with this PR is around exporting a global data struct as part of the API, which as far as I understand is bad practice. I know that is how it was before, but I was _ok_ with it at best when we had it unexported. 

Based on what I see in the code however, we just need to pass the parameters for the KubeConfig function in (or maybe even just pass in the `RESTClientGetter`, it will be created by default if `nil` is passed) and `Debug` can just be put back in `cmd` and be unexported. You could still have the global settings object unexported like before in the `cmd` package. Any thoughts? I'd be willing to pair with you if needed as well

**Comment by aaronmell** [comment]
@thomastaylor312 I think I understand what your asking. The second commit I made felt wrong, so I'll back it out, and give it another go.

**Comment by thomastaylor312** [comment]
Thanks @aaronmell. Like I said, if you need a second set of eyes, let me know

**Comment by aaronmell** [comment]
@thomastaylor312 how does it look now?

File: pkg/action/action.go
Comment (thomastaylor312) [inline pkg/action/action.go]: Because this type is used in the exported `InitActionConfig`, it should be exported as well. We also might want a more specific name like `DebugLog`

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
OK, just one needed change on this from what I can see. I am going to test it in the meantime to make sure everything is working well!

File: pkg/action/action.go
Comment (aaronmell) [reply pkg/action/action.go]: Changes Made

**Review by thomastaylor312 — APPROVED** [review]
Thanks for all of the hard work on this @aaronmell! I'll get one of the other maintainers to review as well due to this being marked as a large PR

**Comment by aaronmell** [comment]
@thomastaylor312  Any update?

**Comment by bacongobbler** [comment]
All of us were away for Helm Summit EU this week. Should be back by next week to take another look

**Comment by aaronmell** [comment]
@bacongobbler Awesome, Thanks!

**Comment by aaronmell** [comment]
Any chance of getting a review this week?

**Comment by bacongobbler** [comment]
throwing this on the 3.0.0-beta.4 milestone so we don't miss this.

Sorry for the delay on a review @aaronmell. Right now we're busy reviewing the last few PRs for Helm 2.15. Once that is complete, we can circle back to this PR, merge it, then start the release process for 3.0.0-beta.4.

File: pkg/action/action.go
Comment (mattfarina) [inline pkg/action/action.go]: Helm moved to Go modules. That means the major version needs to be in the import path. This applies when something has a major version of 2 or higher. So, the helm import paths need to have a `/v3` in them. For example,

`"helm.sh/helm/pkg/cli"` -> `"helm.sh/helm/v3pkg/cli"`

Comment (aaronmell) [reply pkg/action/action.go]: I rebased with the latest. Should be good now.

File: pkg/action/action.go
Comment (bacongobbler) [inline pkg/action/action.go]: Can you make this part of the `*cli.EnvSettings` object? e.g. in `pkg/cli`:

```
func (e *EnvSettings) Namespace() string {
    if e.Namespace != "" {
        return e.Namespace
    }

    if ns, _, err := e.kubeConfig().ToRawKubeConfigLoader().Namespace(); err != nil {
        return ns
    }

    return "default"
}
```

File: pkg/action/action.go
Comment (bacongobbler) [inline pkg/action/action.go]: Can you make this part of the `*cli.EnvSettings` object? e.g. in `pkg/cli`:

```
func (e *EnvSettings) KubeConfig() genericclioptions.RESTClientGetter {
    configOnce.Do(func() {
        config = kube.GetConfig(envSettings.KubeConfig, envSettings.KubeContext, envSettings.Namespace)
    })
    return config
}
```

Then `InitActionConfig` can call `envSettings.KubeConfig()`.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/action/action.go
Comment (aaronmell) [reply pkg/action/action.go]: I think I captured the intent here.

File: pkg/action/action.go
Comment (aaronmell) [reply pkg/action/action.go]: I think I captured the intent here

File: go.mod
Comment (bacongobbler) [inline go.mod]: looks like the go.mod file sneaked in a few accidental adds. Would you mind reverting these?

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go]: can you fix up these package imports?

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go]: This change makes `namespace` a shared global variable between all EnvSettings objects. Is there a reason this was refactored out of the EnvSettings object?

This should be an internal variable of the EnvSettings object. e.g.

```
// EnvSettings describes all of the environment settings.
type EnvSettings struct {
	// namespace is the namespace scope.
	namespace string
```

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go]: Same here RE: making this a shared global variable. This should be part of the EnvSettings object as it was before, but as an internal variable.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/cli/environment.go
Comment (aaronmell) [reply pkg/cli/environment.go]: Done

**Review by bacongobbler — APPROVED** [review]
looking muuuuuuuuch better. Thank you!

I noticed that @thomastaylor312 self-requested another review so I'll hold off merging for now and let him merge after re-reviewing this again.

**Comment by bacongobbler** [comment]
Once last CI error:

> GO111MODULE=on /go/bin/golangci-lint run
> pkg/cli/environment.go:32: File is not `goimports`-ed with -local helm.sh/helm (goimports)

> make: *** [Makefile:85: test-style] Error 1

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go]: here's the line failing CI. This should be moved up beside the pflag import.

General practice for Go import statements is

```
import (
    // standard libraries

    // third-party packages

    // local packages
)
```

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/cli/environment.go
Comment (aaronmell) [reply pkg/cli/environment.go]: Yeah, It took me a bit to figure that out. Running make format fixed it up for me.

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go]: Okay, one last thing I'm noticing as I'm thinking through this design.

Given the following use case:

```
s := New()
s.KubeContext = "foo"
s2 := New()
s2.KubeContext = "bar"
...
// insert other setup code required here
```

With this in mind, I'd expect that the following would result in two unique kubeconfigs:

```
s.KubeConfig()
s2.KubeConfig()
```

Instead, what we'll end up with is the same kube config. This is because they are both using shared global flags.

Can we make `configOnce` and `config` part of the EnvSettings object as well? Thanks!

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Review by thomastaylor312 — APPROVED** [review]

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**
